go/arrow/flight/flightsql/driver/driver.go

```diff
@@ -461,43 +480,63 @@ func (c *Connection) QueryContext(ctx context.Context, query string, args []driv
 		return nil, err
 	}
 
-	rows := Rows{}
-	for _, endpoint := range info.Endpoint {
-		schema, records, err := readEndpoint(ctx, c.client, endpoint)
-		if err != nil {
-			return &rows, err
-		}
-		if rows.schema == nil {
-			rows.schema = schema
-		}
-		rows.records = append(rows.records, records...)
-	}
+	rows := newRows()
+	go rows.streamRecordset(ctx, c.client, info.Endpoint)
```

Comment:
should probably use `context.WithCancel` on this and store the cancel function so that `Rows` can call it on `Close` in and tell the streaming it can end early.

Reply:
this is the idea that helped me to better resolve the synchronization.
that's why I dropped the mutex, fearing no deadlocks.
thanks, it improves a lot the PR.